services/src/main/java/edu/unc/lib/dl/cdr/services/rest/modify/ExportCsvController.java

```diff
@@ -0,0 +1,261 @@
+
+    @Autowired
+    private ChildrenCountService childrenCountService;
+
+    protected SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd kk:mm:ss");
+
+    @RequestMapping(value = "{pid}", method = RequestMethod.GET)
+    public ResponseEntity<Map<String, Object>> export(@PathVariable("pid") String pidString, HttpServletRequest request,
+                                                      HttpServletResponse response) throws IOException {
+
+        Map<String, Object> result = new HashMap<>();
+        result.put("action", "exportCsv");
+        result.put("username", request.getRemoteUser());
+
+        try {
+            if (pidString.equals("collections")) {
+                throw new FedoraException("Not allowed to download repository root");
+            }
+
+            PID pid = PIDs.get(pidString);
+
+            SearchRequest searchRequest = generateSearchRequest(request, searchStateFactory.createSearchState());
+            searchRequest.setRootPid(pid);
+            searchRequest.setApplyCutoffs(false);
+
+            SearchState searchState = searchRequest.getSearchState();
+            searchState.setResultFields(Arrays.asList(SearchFieldKeys.ID.name(), SearchFieldKeys.TITLE.name(),
+                    SearchFieldKeys.RESOURCE_TYPE.name(), SearchFieldKeys.ANCESTOR_IDS.name(),
+                    SearchFieldKeys.STATUS.name(), SearchFieldKeys.DATASTREAM.name(),
+                    SearchFieldKeys.ANCESTOR_PATH.name(), SearchFieldKeys.CONTENT_MODEL.name(),
+                    SearchFieldKeys.DATE_ADDED.name(), SearchFieldKeys.DATE_UPDATED.name(),
+                    SearchFieldKeys.LABEL.name(), SearchFieldKeys.CONTENT_STATUS.name()));
+            searchState.setSortType("export");
+            searchState.setRowsPerPage(searchSettings.maxPerPage);
+
+            BriefObjectMetadata container = queryLayer.addSelectedContainer(pid, searchState, false,
+                    searchRequest.getAccessGroups());
+            SearchResultResponse resultResponse = queryLayer.getSearchResults(searchRequest);
+
+            List<BriefObjectMetadata> objects = resultResponse.getResultList();
+            objects.add(0, container);
+
+            childrenCountService.addChildrenCounts(objects, searchRequest.getAccessGroups());
+
+            String filename = pid.getId().replace(":", "_") + ".csv";
+            response.addHeader("Content-Disposition", "attachment; filename=\"" + filename + "\"");
+            response.addHeader("Content-Type", "text/csv");
+
+            try (ServletOutputStream out = response.getOutputStream()) {
+                Writer writer = new BufferedWriter(new OutputStreamWriter(out, StandardCharsets.UTF_8));
+
+                try (CSVPrinter printer = getPrinter(writer)) {
+                    for (BriefObjectMetadata object : objects) {
+                        printObject(printer, object);
+                    }
+                }
+            }
+        } catch (FedoraException e) {
```

Comment:
After making the suggested change about returning a response at line 108 you shouldn't need the `FedoraException` catch anymore

Reply:
One other suggestion generally with errors, if the issue is a client error we probably don't need to dump a stack trace at error level. So for invalid ids, access control errors, resources not found, we would probably either want to log nothing or do it at debug level.